### PR TITLE
Add a ResumeFlowRun automations action

### PR DESCRIPTION
**Pull request by abrookins**
We are currently lacking a `ResumeFlowRun` action that can resume paused and suspended flow runs. Adding this would enable automations that are not currently possible.

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] If this pull request adds new functionality, it includes unit tests that cover the changes
- [ ] If this pull request removes docs files, it includes redirect settings in `mint.json`.
- [x] If this pull request adds functions or classes, it includes helpful docstrings.

### CODE REVIEW

**Review by desertaxle — APPROVED** [review]
LGTM!

File: src/prefect/server/events/actions.py [hunk 1]
```diff
@@ -1083,6 +1087,40 @@ async def new_state(self, triggered_action: "TriggeredAction") -> StateCreate:
...
+        self._resulting_related_resources.append(
+            RelatedResource.model_validate(
+                {
+                    "prefect.resource.id": f"prefect.flow-run.{flow_run_id}",
+                    "prefect.resource.role": "target",
+                }
+            )
+        )
+
+        logger.info(
```
Comment (cicdw) [inline src/prefect/server/events/actions.py:1107]: tiny preference for this to be DEBUG level
```suggestion
        logger.debug(
```

**Review by cicdw — COMMENTED** [review]
sweet thank you! one tiny request that the log be DEBUG instead of INFO

File: src/prefect/server/events/actions.py (hunk 1 above)
Comment (abrookins) [reply src/prefect/server/events/actions.py:1107]: Will do!

**Review by cicdw — APPROVED** [review]

**Review by chrisguidry — APPROVED** [review]

**Pull request merged**